<?xml version="1.0" ?>
<robot name="puzzlebot">
  <material name="blue">
    <color rgba="0.4 0.6 0.8 0.99"/>
  </material>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".001"/>
      <inertia ixx="0.010609" ixy="0" ixz="0" iyy="0.050409" iyz="0" izz="0.05865"/>
    </inertial>
  </link>
  <link name="front_base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".1"/>
      <inertia ixx="0.000028333" ixy="0" ixz="0" iyy="0.000007533" iyz="0" izz="0.000020867"/>
    </inertial>
  </link>
  <joint name="front_base_joint" type="fixed">
    <origin xyz="0.024 0 0.015" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="front_base" />
  </joint>
  <link name="front_center">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005667" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000016667"/>
    </inertial>
  </link>
  <joint name="front_center_joint" type="fixed">
    <origin xyz="0 0 0.02" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_center" />
  </joint>
  <link name="front_c0">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="front_c0_joint" type="fixed">
    <origin xyz="0 -0.018 0.02" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_c0" />
  </joint>
  <link name="front_c1">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="front_c1_joint" type="fixed">
    <origin xyz="0 0.008 0.02" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_c1" />
  </joint>
  <link name="front_top">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000010833" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.0000010433"/>
    </inertial>
  </link>
  <joint name="front_top_joint" type="fixed">
    <origin xyz="0 0 0.03" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_top" />
  </joint>
  <link name="front_c_left">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="front_c_left_joint" type="fixed">
    <origin xyz="0 0.024 0.02" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_c_left" />
  </joint>
  <link name="front_c_right">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="front_c_right_joint" type="fixed">
    <origin xyz="0 -0.024 0.02" rpy="0 0 0" />
    <parent link="front_base" />
    <child link="front_c_right" />
  </joint>

<!-- Left sides -->
  <link name="left_base">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".1"/>
      <inertia ixx="0.000028333" ixy="0" ixz="0" iyy="0.000007533" iyz="0" izz="0.000020867"/>
    </inertial>
  </link>
  <joint name="left_base_joint" type="fixed">
    <origin xyz="0 0.024 0.015" rpy="0 0 1.57079632679" />
    <parent link="base_link" />
    <child link="left_base" />
  </joint>
  <link name="left_center">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005667" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000016667"/>
    </inertial>
  </link>
  <joint name="left_center_joint" type="fixed">
    <origin xyz="0 0 0.02" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_center" />
  </joint>
  <link name="left_c0">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="left_c0_joint" type="fixed">
    <origin xyz="0 -0.018 0.02" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_c0" />
  </joint>
  <link name="left_c1">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="left_c1_joint" type="fixed">
    <origin xyz="0 0.008 0.02" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_c1" />
  </joint>
  <link name="left_top">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000010833" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.0000010433"/>
    </inertial>
  </link>
  <joint name="left_top_joint" type="fixed">
    <origin xyz="0 0 0.03" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_top" />
  </joint>
  <link name="left_c_left">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="left_c_left_joint" type="fixed">
    <origin xyz="0 0.024 0.02" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_c_left" />
  </joint>
  <link name="left_c_right">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="left_c_right_joint" type="fixed">
    <origin xyz="0 -0.024 0.02" rpy="0 0 0" />
    <parent link="left_base" />
    <child link="left_c_right" />
  </joint>

  <!-- Right sides -->
    <link name="right_base">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".1"/>
      <inertia ixx="0.000028333" ixy="0" ixz="0" iyy="0.000007533" iyz="0" izz="0.000020867"/>
    </inertial>
  </link>
  <joint name="right_base_joint" type="fixed">
    <origin xyz="0 -0.024 0.015" rpy="0 0 -1.57079632679" />
    <parent link="base_link" />
    <child link="right_base" />
  </joint>
  <link name="right_center">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005667" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000016667"/>
    </inertial>
  </link>
  <joint name="right_center_joint" type="fixed">
    <origin xyz="0 0 0.02" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_center" />
  </joint>
  <link name="right_c0">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="right_c0_joint" type="fixed">
    <origin xyz="0 -0.018 0.02" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_c0" />
  </joint>
  <link name="right_c1">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="right_c1_joint" type="fixed">
    <origin xyz="0 0.008 0.02" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_c1" />
  </joint>
  <link name="right_top">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000010833" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.0000010433"/>
    </inertial>
  </link>
  <joint name="right_top_joint" type="fixed">
    <origin xyz="0 0 0.03" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_top" />
  </joint>
  <link name="right_c_left">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="right_c_left_joint" type="fixed">
    <origin xyz="0 0.024 0.02" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_c_left" />
  </joint>
  <link name="right_c_right">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="right_c_right_joint" type="fixed">
    <origin xyz="0 -0.024 0.02" rpy="0 0 0" />
    <parent link="right_base" />
    <child link="right_c_right" />
  </joint>

<!-- Back Side -->
<link name="back_base">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.03" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".1"/>
      <inertia ixx="0.000028333" ixy="0" ixz="0" iyy="0.000007533" iyz="0" izz="0.000020867"/>
    </inertial>
  </link>
  <joint name="back_base_joint" type="fixed">
    <origin xyz="-0.024 0 0.015" rpy="0 0 3.141592653589793238" />
    <parent link="base_link" />
    <child link="back_base" />
  </joint>
  <link name="back_center">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.006 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005667" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000016667"/>
    </inertial>
  </link>
  <joint name="back_center_joint" type="fixed">
    <origin xyz="0 0 0.02" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_center" />
  </joint>
  <link name="back_c0">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="back_c0_joint" type="fixed">
    <origin xyz="0 -0.018 0.02" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_c0" />
  </joint>
  <link name="back_c1">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.0045" length="0.008" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000005198" ixy="0" ixz="0" iyy="0.0000005198" iyz="0" izz="0.0000005063"/>
    </inertial>
  </link>
  <joint name="back_c1_joint" type="fixed">
    <origin xyz="0 0.008 0.02" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_c1" />
  </joint>
  <link name="back_top">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.05 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000010833" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.0000010433"/>
    </inertial>
  </link>
  <joint name="back_top_joint" type="fixed">
    <origin xyz="0 0 0.03" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_top" />
  </joint>
  <link name="back_c_left">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="back_c_left_joint" type="fixed">
    <origin xyz="0 0.024 0.02" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_c_left" />
  </joint>
  <link name="back_c_right">
    <visual>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.002 0.002 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value=".05"/>
      <inertia ixx="0.0000004333" ixy="0" ixz="0" iyy="0.0000004333" iyz="0" izz="0.000000033333"/>
    </inertial>
  </link>
  <joint name="back_c_right_joint" type="fixed">
    <origin xyz="0 -0.024 0.02" rpy="0 0 0" />
    <parent link="back_base" />
    <child link="back_c_right" />
  </joint>


<link name="left_wheel">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.002"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.002"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="0.1" />
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0000025333" ixy="0.0" ixz="0.0" iyy="0.0000025333" iyz="0.0" izz="0.000005"/>
    </inertial>
  </link>

  <link name="right_wheel">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.002"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.01" length="0.002"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="0.1" />
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0000025333" ixy="0.0" ixz="0.0" iyy="0.0000025333" iyz="0.0" izz="0.000005"/>
    </inertial>
  </link>

  <joint name="left_wheel_joint" type="continuous">
    <origin rpy="-1.57079632679 0 0" xyz="0 -0.02 0.002" />
    <parent link="base_link" />
    <child link="left_wheel" />
    <axis xyz="0 0 1" />
  </joint>
  <joint name="right_wheel_joint" type="continuous">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.02 0.002" />
    <parent link="base_link" />
    <child link="right_wheel" />
    <axis xyz="0 0 1" />
  </joint>
  <transmission name="left_wheel_transmission" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_wheel_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="right_wheel_transmission" type="SimpleTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_wheel_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <link name="front_wheel">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="0.1" />
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.00000010667" ixy="0.0" ixz="0.0" iyy="0.00000010667" iyz="0.0" izz="0.00000010667"/>
    </inertial>
  </link>
  <link name="back_wheel">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="0.1" />
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.00000010667" ixy="0.0" ixz="0.0" iyy="0.00000010667" iyz="0.0" izz="0.00000010667"/>
    </inertial>
  </link>

  <joint name="front_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.02 0 -0.00397" />
    <parent link="base_link" />
    <child link="front_wheel" />
  </joint>
  <joint name="back_wheel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.02 0 -0.00397" />
    <parent link="base_link" />
    <child link="back_wheel" />
  </joint>
</robot>
